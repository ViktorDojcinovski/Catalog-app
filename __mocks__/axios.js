const axiosMock = jest.genMockFromModule('axios');

let mockResponse = {
  data: {
    "catalogues": [{
      "name": "Test name",
      "type": "Test type"
    }]
  }
}

axiosMock.get.mockImplementation(req);

function req() {
  return new Promise(function(resolve) {
    axiosMock.delayTimer = setTimeout(function() {
      resolve(mockResponse);
    }, 1000)
  })
}

module.exports = axiosMock;
