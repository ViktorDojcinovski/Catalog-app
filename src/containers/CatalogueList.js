import React, { Component } from "react";
import styled from "styled-components";
import axios from "axios";

import CatalogueAbstract from "../components/CatalogueAbstract";

const StyledWrapper = styled.section`
  overflow: auto;
  background-color: rgba(0, 0, 0, 0.1);
  padding: 10px;
  text-align: center;
`;

class CatalogueList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      catalogues: []
    };
  }
  async componentDidMount() {
    let response = await axios.get(
      "http://localhost:3000/data/catalogues.json"
    );
    this.setState({ catalogues: response.data.catalogues });
  }
  render() {
    let catalogAbstracts = this.state.catalogues.map((catalogue, id) => {
      return (
        <CatalogueAbstract
          key={id}
          name={catalogue.name}
          type={catalogue.type}
        />
      );
    });
    return <StyledWrapper>{catalogAbstracts}</StyledWrapper>;
  }
}

export default CatalogueList;
