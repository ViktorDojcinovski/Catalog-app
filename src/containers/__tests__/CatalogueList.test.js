import React from "react";
import { shallow } from "enzyme";
import axios from "axios";
import renderer from "react-test-renderer";

import CatalogueList from "../CatalogueList";

describe("Catalogue List section", function() {
  let mountedCatalogueList;
  beforeEach(() => {
    mountedCatalogueList = shallow(<CatalogueList />);
  });
  it("renders correctly", () => {
    const tree = renderer.create(<CatalogueList />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders without crashing", () => {
    shallow(<CatalogueList />);
  });
  it("calls axios.get in #componentDidMount", () => {
    return mountedCatalogueList
      .instance()
      .componentDidMount()
      .then(() => {
        expect(axios.get).toHaveBeenCalled();
      });
  });
  it("calls axios.get with correct url", () => {
    return mountedCatalogueList
      .instance()
      .componentDidMount()
      .then(() => {
        expect(axios.get).toHaveBeenCalledWith(
          "http://localhost:3000/data/catalogues.json"
        );
      });
  });
  it("updaes the state with the data from the api endpoint", () => {
    return mountedCatalogueList
      .instance()
      .componentDidMount()
      .then(() => {
        expect(mountedCatalogueList.state()).toHaveProperty("catalogues", [
          {
            name: "Test name",
            type: "Test type"
          }
        ]);
      });
  });
});
