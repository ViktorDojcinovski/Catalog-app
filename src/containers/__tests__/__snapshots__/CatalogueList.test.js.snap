// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalogue List section renders correctly 1`] = `
<section
  className="sc-bxivhb fPyRTT"
/>
`;
